python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -257,8 +257,9 @@ def col(col: str) -> Column:
 column = col
 
 
-def lit(col: Any) -> Column:
+def lit(col: Any, to_struct: bool = False) -> Column:
```

Comment:
I think we'd better make it respect `"spark.sql.pyspark.inferNestedDictAsStruct.enabled"`

instead of adding a new parameter

Reply:
so I get red of the param and add this

sc = _get_active_spark_context()
is_dict_inferred_to_struct = sc._conf.get(“spark.sql.pyspark.inferNestedDictAsStruct”)
# Convert to struct or map based on the parameter `to_struct`
if is_dict_inferred_to_struct:
    return struct(*[lit(value).alias(key) for key, value in col.items()])
else:
    return create_map(*[lit(x) for x in chain(*col.items())])
    
    
   what do you think ?